### PR TITLE
Make paraswap slippage configurable

### PR DESCRIPTION
**Pull request by fleupold**
May fix an alert we have seen somewhat frequently in prod:

> ERROR (pod: dfusion-v2-solver-mainnet-75875777c5-gzjq5):
2021-06-23T12:18:13.603Z ERROR solver::solver::single_order_solver: Inner solver error: TransactionBuilderQuery result parsing failed: {"error":"Destination Amount Mismatch"}
Caused by:
   missing field from at line 1 column 39

My assumption is that slippage is changing the destination amount too much, so that the order is no longer fillable. However, the [docs for the transaction builder query](https://developers.paraswap.network/api/build-parameters-for-transaction-builder-only) specifically state that the `destAmount` can be adjusted with slippage. My strategy would be to try setting slippage to 0 and if we see too many failed settlement we can increase it back to a value that we feel comfortable with.

In that case we might want to start treating the error above as "benign" and not alert on it.

### Test Plan
Added a unittest to verify slippage computation logic.

### CODE REVIEW

File: solver/src/main.rs [hunk 1]
```diff
@@ -148,6 +148,10 @@ struct Arguments {
         parse(try_from_str = shared::arguments::wei_from_gwei)
     )]
     gas_price_cap: f64,
+
+    /// The maximum gas price the solver is willing to pay in a settlement
+    #[structopt(long, env, default_value = "0")]
+    paraswap_slippage_bps: usize,
```
Comment (vkgnosis) [inline solver/src/main.rs:154]: Is this more natural as a bps, percentage or float ratio? 1 bps = 0.01% = 0.0001 (unit less).
I personally prefer the ratio but I realize it can be hard to read when there are too many zeroes and in finance maybe bps is more common.

**Review by vkgnosis — APPROVED** [review]

**Review by bh2smith — APPROVED** [review]

File: solver/src/main.rs (hunk 1 above)
Comment (fleupold) [reply solver/src/main.rs:154]: I think bps is a quite common unit in finance. The problem with floats is that the conversion to u256 and back may be lossy (but I'm also happy to pass it as a float). Ratios are quite hard to parse from env variable aren't they?

Comment (nlordell) [reply solver/src/main.rs:154]: `BigDecimal` is also possible as it has a `FromStr` implementation (although might be overkill).

**Review by nlordell — APPROVED** [review]

**Comment by josojo** [comment]
If this happens mostly for buy orders, maybe also this issue gives a hint: https://github.com/gnosis/gp-v2-services/issues/788
